MibTeX/src/de/mibtex/export/typo3/Typo3Entry.java

```diff
@@ -0,0 +1,185 @@
+		ToURLOverwrites.put("&8211;", "-");
+		ToURLOverwrites.put("?", "?");
+		ToURLOverwrites.put(":", ":");
+		ToURLOverwrites.put("/", "/");
+		ToURLOverwrites.put("#", "#");
+		ToURLOverwrites.put("&szlig;", "�");
+	}
+	
+	public final BibtexEntry source;
+	
+	public String type; // inproceedings, article, ...
+	public String key;
+	
+	public String typeAttrib; // type = {...}
+	public String title;
+	public List<String> authors;
+	public List<String> editors;
+	public String booktitle;
+	public String address;
+	public String publisher;
+	public String journal;
+	public String location;
+	public String school;
+	public String pages;
+	public int year;
+	
+	public String doi;
+	public String isbn;
+	public String issn;
+	
+	public String note;
+	
+	public List<String> tags;
+	
+	public Typo3Entry(BibtexEntry bib, Map<String, String> variables) {
+		this.source = bib;
+		
+		this.type = bib.type;
+		this.typeAttrib = MakeTypo3Safe(bib.getAttribute(BibTeXEntry.KEY_TYPE));
+		this.key = bib.key;
+
+		this.authors = new ArrayList<>();
+		this.editors = new ArrayList<>();
+		List<String> persons = bib.authorList.stream().map(Typo3Entry::MakeTypo3Safe).collect(Collectors.toList());
+		if (bib.authorsAreEditors) {
+			this.editors = persons;
+		} else {
+			this.authors = persons;
+		}
+		
+		this.title = MakeTypo3Safe(bib.title);
+		this.year = bib.year;
+
+		this.address = MakeTypo3Safe(lookup(bib.getAttribute(BibTeXEntry.KEY_ADDRESS), variables));
+		this.publisher = MakeTypo3Safe(lookup(bib.getAttribute(BibTeXEntry.KEY_PUBLISHER), variables));
+		this.journal = MakeTypo3Safe(lookup(bib.getAttribute(BibTeXEntry.KEY_JOURNAL), variables));
+		this.location = MakeTypo3Safe(lookup(bib.getAttribute("location"), variables));
+		
+		this.school = MakeTypo3Safe(bib.getAttribute(BibTeXEntry.KEY_SCHOOL));
+		this.pages = MakeTypo3Safe(bib.getAttribute(BibTeXEntry.KEY_PAGES));
+		
+		this.doi = MakeTypo3Safe(bib.getAttribute(BibTeXEntry.KEY_DOI));
+		this.isbn = bib.getAttribute("isbn");
+		this.issn = bib.getAttribute("issn");
+		
+		this.note = MakeTypo3Safe(bib.getAttribute(BibTeXEntry.KEY_NOTE));
+		
+		{
+			String booktitle;
+			if (Filters.Is_techreport.test(this)) {
+				booktitle = ("Technical Report " + bib.getAttribute(BibTeXEntry.KEY_NUMBER)).trim();
+//			} else if (Filters.is_phdthesis.test(bib)) {
+//				booktitle = "PhD Thesis";
+//			} else if (Filters.is_bachelorsthesis.test(bib)) {
+//				booktitle = "Bachelor's Thesis";
+//			} else if (Filters.is_mastersthesis.test(bib)) {
+//				booktitle = "Master's Thesis";
+			} else {
+				booktitle = lookup(bib.getAttribute(BibTeXEntry.KEY_BOOKTITLE), variables);
+			}
+			this.booktitle = MakeTypo3Safe(booktitle);
+		}
+		
+		this.tags = bib.tagList.get(ExportTypo3Bibtex.Typo3TagsAttribute);
+		if (this.tags == null) {
+			this.tags = new ArrayList<>();
+		} else {
+			this.tags = Util.SplitAttributeListString(this.tags);
+		}
+	}
+
+	@Override
+	public String toString() {
```

Comment:
Split long method

Reply:
Extracted the tedious creation of the author attribute but left the rest because I don't see how and why splitting the string conversion of the other attributes could yield a benefit.

Reply:
see above, these are very generic common guidelines

Reply:
Each of the 19 attributes of this class has to be put into the string. Each attribute takes only a single line now. I have no idea how to shorten this more. Abstracting over fields is not possible in Java without reflection which would be an overkill here I guess (to loop over fields instead of having one line for each of them with almost similar content).